typescript/infra/scripts/send-random-messages.ts

```diff
@@ -0,0 +1,183 @@
+  Chains,
+  HookType,
+  HyperlaneCore,
+  MultiProvider,
+} from '@hyperlane-xyz/sdk';
+import { addressToBytes32, sleep } from '@hyperlane-xyz/utils';
+
+const ANVIL_KEY =
+  '0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80';
+
+enum MailboxHookType {
+  REQUIRED = 'requiredHook',
+  DEFAULT = 'defaultHook',
+}
+
+/**
+ * If a hookArg is provided, set the mailbox hook to the defaultHookArg.
+ * The hook is set either as the default hook or the required hook,
+ * depending on the mailboxHookType argument.
+ */
+async function setMailboxHook(
+  mailbox: Mailbox,
+  coreAddresses: any,
+  local: ChainName,
+  mailboxHookType: MailboxHookType,
+  hookArg: HookType,
+) {
+  const hook = coreAddresses[local][hookArg];
+  switch (mailboxHookType) {
+    case MailboxHookType.REQUIRED: {
+      await mailbox.setRequiredHook(hook);
+      break;
+    }
+    case MailboxHookType.DEFAULT: {
+      await mailbox.setDefaultHook(hook);
+      break;
+    }
+  }
+  console.log(`set the ${mailboxHookType} hook on ${local} to ${hook}`);
+}
+
+const chainSummary = async (core: HyperlaneCore, chain: ChainName) => {
+  const coreContracts = core.getContracts(chain);
+  const mailbox = coreContracts.mailbox;
+  const dispatched = await mailbox.nonce();
+  // TODO: Allow processed messages to be filtered by
+  // origin, possibly sender and recipient.
+  const processFilter = mailbox.filters.Process();
+  const processes = await mailbox.queryFilter(processFilter);
+  const processed = processes.length;
+
+  const summary = {
+    chain,
+    dispatched,
+    processed,
+  };
+  return summary;
+};
+
+function getArgs() {
+  return yargs(process.argv.slice(2))
+    .option('messages', {
+      type: 'number',
+      describe: 'Number of messages to send; defaults to having no limit',
+      default: 0,
+    })
+    .option('timeout', {
+      type: 'number',
+      describe: 'Time to wait between messages in ms.',
+      default: 5000,
+    })
+    .option('mineforever', {
+      type: 'boolean',
+      default: false,
+      describe: 'Mine forever after sending messages',
+    })
+    .option(MailboxHookType.DEFAULT, {
+      type: 'string',
+      describe: 'Description for defaultHook',
+      choices: Object.values(HookType),
+      default: HookType.AGGREGATION,
+    })
+    .option(MailboxHookType.REQUIRED, {
+      type: 'string',
+      describe: 'Required hook to call in postDispatch',
+      choices: Object.values(HookType),
+      default: HookType.PROTOCOL_FEE,
+    }).argv;
+}
+
+async function main() {
+  const args = await getArgs();
+  const { timeout, defaultHook, requiredHook, mineforever } = args;
+  let messages = args.messages;
+
+  const signer = new Wallet(ANVIL_KEY);
+  const multiProvider = MultiProvider.createTestMultiProvider({ signer });
+  const provider = multiProvider.getProvider(Chains.test1);
+
+  const addresses = JSON.parse(
+    fs.readFileSync('./config/environments/test/core/addresses.json', 'utf8'),
+  );
```

Comment:
surprised this just works? where is this deployed

Reply:
This file was the HH task formerly known as Kathy which the e2e agent test uses. I'll rename it for clarity